index.bs

```diff
@@ -4572,6 +4572,55 @@ and similarly, closing or aborting the <code>writable</code> side will implicitl
 
 </div>
 
+<h3 id="example-ts-promise-to-readable">Using an identity transform stream as a primitive when manipulating streams</h3>
+
+Combining an identity transform stream with {{pipeTo()}} is a powerful way to manipulate streams. For example,
+concatenating multiple readable streams into one:
+
+<pre><code class="lang-javascript">
+function concatenateReadables(readables) {
+  const ts = new TransformStream();
+  let promise = Promise.resolve();
+  for (const readable of readables) {
+    promise = promise
+        .then(() => readable.pipeTo(ts.writable, { preventClose: true }),
+              reason => {
+                return Promise.all([
+                  ts.writable.abort(reason),
+                  readable.cancel(reason)]);
+              });
+  }
+  promise.then(() => ts.writable.getWriter().close(),
+               reason => ts.writable.abort(reason))
+      .catch(() => {});
+  return ts.readable;
+}
+</code></pre>
+
+The error handling here is subtle because cancelling the concatenated stream has to cancel all the intput
+streams. However, the success case is simple enough. We just pipe each stream in the <code>readables</code> array one at
+a time to the transform stream's <a>writable side</a>, and then close it when we are done. The <a>readable side</a> is
+then a concatenation of all the chunks from all of of the streams. We return it from the function. Backpressure is
+applied as usual.
+
+It's sometimes natural to treat a promise for a readable as if it were a readable. A simple adaptor function is all
+that's needed:
+
+<pre><code class="lang-javascript">
+function promiseToReadable(promiseForReadable) {
+  const ts = new TransformStream();
+  promiseForReadable.then(readable => readable.pipeTo(ts.writable))
+      .catch(reason => ts.writable.abort(reason))
```

Comment:
Since we didn't pass preventAbort: true, why is this necessary?

Reply:
It can fail if `readable` doesn't implement pipeTo(), or if pipeTo() is implemented but throws, or readable is locked... probably some more cases too.